### ISSUE TITLE
okhttp-tests failure running clean build

### ISSUE THREAD

**Issue by mdsiebler** [square/okhttp#3588]
I am trying to verify my build of the okhttp library. Running 'mvn -X clean verify' on a recent download of the master branch. I get the following error: It is reproducible. Is this a known problem with the testbase? FYI this is version 3.8.1 of the library. Thanks

P.S. - if there is more data I can supply I am happy to provide. I am new to github so I am not sure how much detail I should provide. I apologize in advance if I am not following the community rules. Thanks in advance.

[matthias@oc0105085538 surefire-reports]$ more okhttp3.ConnectionCoalescingTest.txt
-------------------------------------------------------------------------------
Test set: okhttp3.ConnectionCoalescingTest
-------------------------------------------------------------------------------
Tests run: 12, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 5.237 sec <<< FAILURE! - in okhttp3.ConnectionCoalescingTest
prefersExistingCompatible(okhttp3.ConnectionCoalescingTest)  Time elapsed: 0.598 sec  <<< FAILURE!
java.lang.AssertionError: expected:<3> but was:<2>
    at org.junit.Assert.fail(Assert.java:88)
    at org.junit.Assert.failNotEquals(Assert.java:834)
    at org.junit.Assert.assertEquals(Assert.java:645)
    at org.junit.Assert.assertEquals(Assert.java:631)
    at okhttp3.ConnectionCoalescingTest.prefersExistingCompatible(ConnectionCoalescingTest.java:256)

**Comment by dave-r12** [comment]
Have you altered the code? I see this on line 256 for 3.8.1:
```java
assertEquals(1, client.connectionPool().connectionCount());
```

I think we fixed this with 3.9.0. Can you try with the latest?

**Comment by mdsiebler** [comment]
Thanks this was indeed fixed in 3.9.0
